hypershift/hosted-wrapper.py

```diff
@@ -0,0 +1,592 @@
+                metadata['cluster_name'] = cluster['infra_id']
+                metadata['cluster_id'] = cluster['id']
+                metadata['version'] = cluster['openshift_version']
+                metadata['base_domain'] = cluster['dns']['base_domain']
+                metadata['aws_region'] = cluster['region']['id']
+                metadata['workers'] = cluster['nodes']['compute']
+                metadata['workers_type'] = cluster['nodes']['compute_machine_type']['id']
+                metadata['network_type'] = cluster['network']['type']
+                metadata["timestamp"] = time.strftime("%Y-%m-%dT%H:%M:%S")
+                metadata['hostedclusters'] = hostedclusters
+                metadata['workers_per_nodepool'] = hosted_workers
+                es_ignored_metadata = ""
+                common._index_result(es,index,metadata,es_ignored_metadata,index_retry)
+                return metadata
+
+
+def _download_kubeconfig(ocm_cmnd,mgmt_cluster_id,my_path):
+    logging.debug('Downloading kubeconfig file for Management Cluster %s on %s' % (mgmt_cluster_id,my_path))
+    kubeconfig_cmd = ["ocm", "get", "/api/clusters_mgmt/v1/clusters/" + mgmt_cluster_id + "/credentials"]
+    logging.debug(kubeconfig_cmd)
+    process = subprocess.Popen(kubeconfig_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,cwd=my_path,universal_newlines=True)
+    stdout,stderr = process.communicate()
+    if process.returncode != 0:
+        logging.error('Failed to download kubeconfig file for Management Cluster ID %s with this stdout/stderr:' % mgmt_cluster_id)
+        logging.error(stdout)
+        logging.error(stderr)
+    else:
+        kubeconfig_path = my_path + "/mgmt_kubeconfig"
+        with open(kubeconfig_path, "w") as kubeconfig_file:
+            kubeconfig_file.write(json.loads(stdout)['kubeconfig'])
+        logging.debug('Downloaded kubeconfig file for Management Cluster ID %s and stored at %s' % (mgmt_cluster_id, kubeconfig_path))
+        return kubeconfig_path
+
+
+def _build_cluster(hypershift_cmnd, kubeconfig_location, cluster_name_seed, mgmt_cluster_base_domain, worker_nodes, mgmt_cluster_aws_zone, pull_secret_file, my_path, my_uuid, my_inc, es, index, index_retry, mgmt_cluster_name):
+    os.environ["KUBECONFIG"] = kubeconfig_location
+    # pass that dir as the cwd to subproccess
+    cluster_path = my_path + "/" + cluster_name_seed + "-" + str(my_inc).zfill(4)
+    os.mkdir(cluster_path)
+    logging.debug('Attempting cluster installation')
+    logging.debug('Output directory set to %s' % cluster_path)
+    cluster_name = cluster_name_seed + "-" + str(my_inc).zfill(4)
+    cluster_cmd = [hypershift_cmnd, "create","cluster", "aws", "--name", cluster_name, "--base-domain", mgmt_cluster_base_domain, "--additional-tags", "mgmt-cluster=" + mgmt_cluster_name, "--aws-creds", my_path + '/aws_creds', "--pull-secret", pull_secret_file, "--region", mgmt_cluster_aws_zone, "--node-pool-replicas", worker_nodes, '--wait']
```

Comment:
We had this similar problem when creating the script for rosa. We only add as parameters the ones we need to manage inside the script and could be wrong. If the only thing we are going to do with the parameter, is to directly pass to the command execution, it has to go to the wildcard option

Reply:
But these `--control-plane-availability-policy`, `--network-type` are mandatory in our perfscale tests. 
Do we have wildcard option in current implementation ?

Reply:
Both parameters can be include on the wildcard parameter as 
`--wildcard-options "--network-type OVNKubernetes --control-plane-availability-policy HighlyAvailable"`

It will result on following hypershift command:

_build_cluster: ['/tmp/xxxxx/hypershift/bin/hypershift', 'create', 'cluster', 'aws', '--name', 'xxxx', '--base-domain', 'xxxxx', '--additional-tags', 'mgmt-cluster=perf-411-f378-btdzf', '--aws-creds', '/tmp/xxxxx/aws_creds', '--pull-secret', '/home/morenod/Downloads/pull-secret.txt', '--region', 'us-west-2', '--node-pool-replicas', '1', '--wait', '--network-type', 'OVNKubernetes', '--control-plane-availability-policy', 'HighlyAvailable']